conan/internal/api/detect_api.py

```diff
@@ -122,7 +122,7 @@ def _get_e2k_architecture():
     }.get(platform.processor())
 
 
-def detect_libcxx(compiler, version):
+def detect_libcxx(compiler, version, compiler_exe):
     assert isinstance(version, Version)
 
     def _detect_gcc_libcxx(version_, executable):
```

Comment:
This function checks for `if executable == "g++": ` to do some special handling, but now the executable can be whatever. Should we keep this around?